backend/migrations/20230310_000000_migration.sql

```diff
@@ -0,0 +1,284 @@
+
+CREATE TYPE scheduling_rule_type AS ENUM ('pause', 'block');
+
+CREATE TABLE IF NOT EXISTS
+scheduling_rules_v0
+( id SERIAL PRIMARY KEY
+, rule_type scheduling_rule_type NOT NULL
+, canvas_id UUID REFERENCES canvases_v0(id) NOT NULL
+, handler_name TEXT NOT NULL
+, event_space TEXT NOT NULL
+, created_at TIMESTAMPTZ NOT NULL DEFAULT now()
+);
+
+
+CREATE TABLE IF NOT EXISTS
+secrets_v0
+( id UUID PRIMARY KEY
+, canvas_id UUID REFERENCES canvases_v0(id) NOT NULL
+, secret_name VARCHAR(255) NOT NULL
+, secret_value TEXT NOT NULL
+, secret_version INT NOT NULL DEFAULT 0
+, created_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+);
+
+
+CREATE TABLE IF NOT EXISTS
+stored_events_v0
+( canvas_id UUID REFERENCES canvases_v0(id) NOT NULL
+, module TEXT NOT NULL
+, path TEXT NOT NULL
+, modifier TEXT NOT NULL
+, timestamp TIMESTAMPTZ NOT NULL
+, value TEXT NOT NULL
+, trace_id UUID NOT NULL
+);
+
+CREATE INDEX IF NOT EXISTS
+idx_stored_events_v0_most_recent
+ON stored_events_v0
+(canvas_id, module, path, modifier, timestamp DESC);
+
+CREATE INDEX IF NOT EXISTS
+stored_events_v0_traceid
+ON stored_events_v0
+(canvas_id, trace_id);
+
+CREATE INDEX IF NOT EXISTS
+idx_stored_events_v0_most_recent_with_text
+ON stored_events_v0
+(canvas_id, module, path text_pattern_ops, modifier, "timestamp" DESC);
+
+
+
+CREATE TABLE IF NOT EXISTS
+system_migrations_v0
+( name TEXT PRIMARY KEY
+, execution_date TIMESTAMPTZ NOT NULL
+, sql TEXT NOT NULL
+);
+
+
+
+CREATE TYPE toplevel_type AS
+ENUM ('handler', 'db', 'user_function');
+
+CREATE TABLE IF NOT EXISTS
+toplevel_oplists_v0
+( canvas_id UUID REFERENCES canvases_v0(id) NOT NULL
+, tlid INTEGER NOT NULL
+, digest CHAR(32) NOT NULL
+, tipe toplevel_type
+, name TEXT /* handlers only - used for http lookups */
+, module TEXT /* handlers only */
+, modifier TEXT /* handlers only */
+, updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+, created_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+, deleted BOOLEAN
+, oplist BYTEA
+, oplist_cache BYTEA
+, PRIMARY KEY (canvas_id, tlid)
+);
+
+CREATE TRIGGER set_toplevel_oplist_timestamp
+BEFORE UPDATE ON toplevel_oplists_v0
+FOR EACH ROW
+EXECUTE PROCEDURE trigger_set_timestamp();
+
+
+
+-- Keep track of what traces are available for which handler/function/etc
+CREATE TABLE IF NOT EXISTS
+traces_v0
+( id UUID PRIMARY KEY
+, canvas_id UUID NOT NULL
+-- the handler's (or for a function's default trace, the function's) TLID (used to
+-- store the trace data in Cloud Storage)
+, root_tlid BIGINT NOT NULL
+, trace_id UUID NOT NULL
+, callgraph_tlids BIGINT[] NOT NULL -- functions called during the trace
+);
+
+
+CREATE TABLE IF NOT EXISTS
+user_data_v0
+( id UUID PRIMARY KEY
+, canvas_id UUID REFERENCES canvases_v0(id) NOT NULL
+, table_tlid BIGINT NOT NULL
+, user_version INT NOT NULL
+, dark_version INT NOT NULL
+, data JSONB NOT NULL
+, created_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+, updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+, key TEXT NOT NULL
+);
+
+UPDATE user_data_v0 SET key = id::text;
```

Comment:
Not sure what this is?

Reply:
This sets the value of the key column to be the same as the id column casted as a text. Not sure why we needed it